azure-mgmt-compute/src/main/java/com/microsoft/azure/management/compute/VirtualMachine.java

```diff
@@ -262,6 +262,20 @@
         T withOSDiskName(String name);
     }
 
+    interface DefinitionWithVMSize<T extends DefinitionCreatable> {
+        /**
+         * @param sizeName The name of the size for the virtual machine as text
+         * @return The stage representing creatable VM definition
+         */
+        T withSize(String sizeName);
+
+        /**
+         * @param size A size from the list of available sizes for the virtual machine
+         * @return The stage representing creatable VM definition
+         */
+        T withSize(VirtualMachineSizeTypes size);
+    }
```

Comment:
Currently the 'withSize(..)' setters are optional and default to "Basic_A0".

In portal this is something user needs to explicitly specify, so we need to decide whether to make it required or optional for user.